### DISCUSSION TITLE
available rule triggers and their order

### DISCUSSION THREAD

**Discussion by Gifford47** [arendst/Tasmota#13763] [Q&A] [answered]
I would like to ask you a general question to the rule triggers:
the docs only provide an excerpt of some rule triggers. which scheme is used to create the triggers? how can i get a complete list of all triggers?
What is their order?
assuming the device is in deep-sleep or in booting phase. which triggers are triggered first? when do the sensor triggers occur?

**Comment by ascillato** [comment, marked as answer]
> the docs only provide an excerpt of some rule triggers.

Sorry, no. The docs has All available rule triggers. besides the standards, you can also use the JSON output of events as rule triggers.

>What is their order?

First the ones that said BOOT like POWER1#BOOT
Then SYSTEM#BOOT
Then WIFI#CONNECTED
Then MQTT#CONNECTED
then STATE like POWER1#STATE when it occurs

>assuming the device is in deep-sleep or in booting phase. which triggers are triggered first? when do the sensor triggers occur?

The same.

Sensors will trigger depending on how you wrote the rule. If you put tele-XXXX will trigger every teleperiod. If you put SENSOR_VALUE>something, will trigger every time the sensor reaches that value.

**Comment by sfromis** [comment]
There is no defined order, triggers are events *when it happens*. The docs page has a complete list of triggers built into the rules language. In addition, you can expect triggers where you see JSON published for MQTT, including payloads about sensors and status info. There are few cases of specialized devices having JSON triggers not appearing as MQTT messages, which should be documented for the device, like when you have rotary events. As for sensor triggers, they may be triggered each main cycle of Tasmota, depending on the driver there may not be data available for each cycle.
